src/utilities/network-information-utils.ts

```diff
@@ -5,24 +5,32 @@ import { Navigator } from "../types/navigator";
 // #region Functions
 // -----------------------------------------------------------------------------------------
 
-export default function buildNetworkInformationUtils(navigator: Navigator) {
+export default function buildNetworkInformationUtils(window?: Window) {
+    const navigator: Navigator | undefined = window?.navigator;
+
     return {
         /**
          * Returns a NavigatorConnection object if one exists
          */
         getNavigatorConnection(): NetworkInformation | undefined {
             return (
-                navigator.connection ??
-                navigator.mozConnection ??
-                navigator.webkitConnection ??
+                navigator?.connection ??
+                navigator?.mozConnection ??
+                navigator?.webkitConnection ??
                 undefined
             );
         },
 
         /**
          * Returns a `NetworkConnection` object which is an aggregate of `navigator.connection` and `navigator.onLine`
          */
-        getNetworkConnection() {
+        getNetworkConnection():
+            | (NetworkInformation & { isOnline: boolean })
```

Comment:
Is there an interface that represents this? If so, I'd swap it instead of redefining here

Reply:
Yeah.  I don't think there's a type right now, but probably a good idea to add one in.

Reply:
That was my guess based on the JS Doc - that should be updated too.